pkg/artifactory/resource_artifactory_ldap_setting.go

```diff
@@ -68,9 +68,9 @@ func resourceArtifactoryLdapSetting() *schema.Resource {
 		},
 		"user_dn_pattern": {
 			Type:             schema.TypeString,
-			Required:         true,
+			Optional:         true,
 			ValidateDiagFunc: validation.ToDiagFunc(validation.All(validation.StringIsNotEmpty, validateLdapDn)),
```

Comment:
Does allowing the key but effectively no value make sense?? I guess. If you're going to supply one, it has to be valid.

Please confirm with the user that IF they supply a user dn, then it must be valid

Reply:
Sure. Will check with user about this.